.github/actions/self_managed_create/action.yml

```diff
@@ -0,0 +1,91 @@
+name: Self-managed infrastructure creation
+description: "Create the required infrastructure for a Constellation cluster manually."
+
+inputs:
+  cloudProvider:
+    description: "The cloud provider the test runs on."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Copy Terraform configuration and Constellation config
+      shell: bash
+      working-directory:
+      run : |
+        cp -r ${{ github.workspace }}/cli/internal/terraform/terraform/${{ inputs.cloudProvider }} ${{ github.workspace }}/e2e-infra
+        cp ${{ github.workspace }}/constellation-conf.yaml ${{ github.workspace }}/e2e-infra
+
+    - name: Get CSP image reference
+      id: get_image
+      shell: bash
+      working-directory: ${{ github.workspace }}/e2e-infra
+      run : |
+        echo "image_ref=$(bazel run //hack/image-fetch:image-fetch)" >> $GITHUB_OUTPUT
+
+    - name: Write Terraform variables
+      shell: bash
+      working-directory: ${{ github.workspace }}/e2e-infra
+      run : |
+        echo "name = \"$(yq '.name' constellation-conf.yaml)\"" >> terraform.tfvars
```

Comment:
Thinking about using a `-bash` file for this so the we can link it (maybe even embed it) in the docs more easily and have a single source of truth. Do you think that this is possible?

Reply:
We have many places in our docs where embedding files would be really useful. I don't know if / how this could work, but it would be good. In this specific case though, its not really possible as the `{{ inputs.foo }}` references should not be in the docs, env vars would be a possible workaround